src/mbgl/util/literal.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <utility>
+#include <cstddef>
+
+namespace mbgl {
+
+// Using structs instead of constexpr templates because GCC 4.9 doesn't support these yet.
+
+template <char... Cs>
+struct string_literal {
+    static constexpr const char chars[] = { Cs..., 0 };
+
+    static constexpr const char* value() {
+        return chars;
+    }
+};
+
+template <char... Cs>
+constexpr const char string_literal<Cs...>::chars[];
+
+namespace detail {
+
+using string_fn = const char* (*)();
+
+template <string_fn name, std::size_t... Is>
+string_literal<name()[Is]...> to_string_literal(std::index_sequence<Is...>);
+
+constexpr std::size_t string_length(const char* const str, std::size_t len = 0) {
+    return str[0] ? string_length(str + 1, len + 1) : len;
```

Comment:
nit: `++str, ++len` ?

Reply:
why?

Reply:
also, the pointer is const, so we can't really call `++str`.

Reply:
just more common way to represent iteration, I'm not insisting though - pls consider it as an optional nit :-) why does `str` need to be const?